mpack-instance-manager/src/main/python/instance_manager/mpack-instance-manager.py

```diff
@@ -152,25 +157,24 @@ def main(options, args):
                        components_map=parsed_components_map)
 
   elif action == GET_CONF_DIR_ACTION:
-    print get_conf_dir(mpack=options.mpack, mpack_instance=options.mpack_instance,
+    print json.dumps(get_conf_dir(mpack=options.mpack, mpack_instance=options.mpack_instance,
```

Comment:
The command line will still print an indented json output. That is not what is desirable. The command line should return a human readable output and NOT a JSON structure. 
For example see the output of tree command
```
tree /usr/hwx/mpacks
/usr/hwx/mpacks
├── hdpcore
│   └── 1.0.0-b415
│       ├── app_timeline_server -> /usr/hwx/modules/yarn/3.0.0.0-b167
│       ├── datanode -> /usr/hwx/modules/hdfs/3.0.0.0-b167
│       ├── hadoop_client -> /usr/hwx/modules/hadoop_clients/3.0.0.0-b167
│       ├── historyserver -> /usr/hwx/modules/mapreduce2/3.0.0.0-b167
│       ├── journalnode -> /usr/hwx/modules/hdfs/3.0.0.0-b167
│       ├── mpack.json
│       ├── namenode -> /usr/hwx/modules/hdfs/3.0.0.0-b167
│       ├── nfs_gateway -> /usr/hwx/modules/hdfs/3.0.0.0-b167
│       ├── nodemanager -> /usr/hwx/modules/yarn/3.0.0.0-b167
│       ├── resourcemanager -> /usr/hwx/modules/yarn/3.0.0.0-b167
│       ├── secondary_namenode -> /usr/hwx/modules/hdfs/3.0.0.0-b167
│       ├── timeline_reader -> /usr/hwx/modules/yarn/3.0.0.0-b167
│       ├── yarn_registry_dns -> /usr/hwx/modules/yarn/3.0.0.0-b167
│       ├── zkfc -> /usr/hwx/modules/hdfs/3.0.0.0-b167
│       ├── zookeeper_client -> /usr/hwx/modules/zookeeper_clients/3.4.0.0-b38
│       └── zookeeper_server -> /usr/hwx/modules/zookeeper/3.4.0.0-b38
└── ods
    └── 1.0.0-b279
        ├── hadoop_client -> /usr/hwx/modules/hadoop_clients/3.0.0.0-b166
        ├── hbase_client -> /usr/hwx/modules/hbase_clients/2.0.0.0-b219
        ├── hbase_master -> /usr/hwx/modules/hbase/2.0.0.0-b219
        ├── hbase_regionserver -> /usr/hwx/modules/hbase/2.0.0.0-b219
        ├── mpack.json
        └── zookeeper_client -> /usr/hwx/modules/zookeeper_clients/3.4.0.0-b38
```

The command line should print something like the following 

```
hdpcore
  |__ HDPCORE (default)
      |__ zookeeper
            |__ zookeeper_server (default)
                     |─ mpack_version -> 1.0.0-b415
                     |─ module_version -> 3.4.0.0-b38
                     |─ instance_path -> /usr/hwx/instances/..../current
                     |─ mpack_path -> /usr/hwx/mpacks/....
                     |─ module_path -> /usr/hwx/modules/....
                     |─ conf_dir -> /usr/hwx/instances/..../conf
                     |─ log_dir -> /usr/hwx/instances/..../logs                            
                     |─ run_dir -> /usr/hwx/instances/..../run
```

Reply:
Lets work on the formatting of the command line output in a non-JSON format in a separate JIRA.

Reply:
I agree, this jira is a blocker for some functionality missing in mpack_instance_manager, the format of output is just for cli stdout, the tree-like output will involve more work, I will create another jira for it.

Reply:
A new JIRA: AMBARI-24196 has been filed to address this new requirement

Reply:
OK, I'm fine with that.